### PR TITLE
[TL-27] Filter asset types

### PR DESCRIPTION
**Pull request by alex-demetriou**
## Adds filtering on `AssetType` for housing search queries

`AssetType` is filtered from a comma-separated list provided to the `HousingSearchRequest` in the URL query params.

#### _Checklist_

- [x] Added end-to-end (i.e. integration) tests where necessary e.g to test complete functionality of a newly added feature
- [x] Added comments to the README or updated relevant documentation _(add link to documentation)_, where necessary.
- [x] Checked all code for possible refactoring
- [x] Code pipeline builds correctly

### *Follow up actions after merging PR*

Currently I assign for `AssetType` in `AssetFixture` the same value as is used for  `AssetAddress.Uprn`. 
Might be good for this to be more random.

### CODE REVIEW

File: HousingSearchApi/V1/Interfaces/AssetQueryGenerator.cs [hunk 1]
```diff
@@ -16,18 +19,35 @@ public AssetQueryGenerator(IWildCardAppenderAndPrepender wildCardAppenderAndPrep
 
         public QueryContainer Create(HousingSearchRequest request, QueryContainerDescriptor<QueryableAsset> q)
         {
-            if (string.IsNullOrWhiteSpace(request.SearchText)) return null;
```
Comment (mikemoody-amido) [inline HousingSearchApi/V1/Interfaces/AssetQueryGenerator.cs:19]: Surely you still need to do this? There should always be some search text shouldn't there?

File: HousingSearchApi/V1/Interfaces/AssetQueryGenerator.cs [hunk 2]
```diff
@@ -16,18 +19,35 @@ public AssetQueryGenerator(IWildCardAppenderAndPrepender wildCardAppenderAndPrep
...
+            if (!(request is HousingSearchRequest housingSearchRequest)) return null;
```
Comment (mikemoody-amido) [inline HousingSearchApi/V1/Interfaces/AssetQueryGenerator.cs:22]: Given that the `request` input parameter is strongly typed on the method definition, this check will never fail and so could be removed.

Comment (evak6979) [reply HousingSearchApi/V1/Interfaces/AssetQueryGenerator.cs:22]: This PR needs tests - both E2E and unit.

**Comment by evak6979** [comment]
This PR needs tests - both E2E and unit.

File: HousingSearchApi/V1/Interfaces/AssetQueryGenerator.cs (hunk 1 above)
Comment (alex-demetriou) [reply HousingSearchApi/V1/Interfaces/AssetQueryGenerator.cs:19]: if requests without search text are never valid, would it make sense to instead have the `[Required]` attribute on the Request model?

Comment (alex-demetriou) [reply HousingSearchApi/V1/Interfaces/AssetQueryGenerator.cs:19]: > if requests without search text are never valid, would it make sense to instead have the `[Required]` attribute on the Request model?

I've just noticed we are using `FluentValidator` so a request with empty search text will never even make it to the controller

**Comment by alex-demetriou** [comment]
@evak6979
I've tested the new filter in dev environment - everything seems to be working perfectly
I will re-add the CircleCI branch filter and we should be good to go

File: HousingSearchApi/V1/Interfaces/AssetQueryGenerator.cs (hunk 1 above)
Comment (evak6979) [reply HousingSearchApi/V1/Interfaces/AssetQueryGenerator.cs:19]: It would still be good to do defensive coding. Check where we can. :)

**Review by evak6979 — APPROVED** [review]

**Pull request merged**